lib/post_destroyer.rb

```diff
@@ -221,24 +221,27 @@ def trash_public_post_actions
     end
   end
 
-  def agree_with_flags
-    if @post.has_active_flag? && @user.id > 0 && @user.staff?
-      Jobs.enqueue(
-        :send_system_message,
-        user_id: @post.user.id,
-        message_type: :flags_agreed_and_post_deleted,
-        message_options: {
-          url: @post.url,
-          flag_reason: I18n.t(
-            "flag_reasons.#{@post.active_flags.last.post_action_type.name_key}",
-            locale: SiteSetting.default_locale,
-            base_path: Discourse.base_path
-          )
-        }
-      )
+  def handle_flags(agree: false)
```

Comment:
I personally think we should still keep `agree_with_flags` because the method name is clear and easy to reason about. What we can do here instead is to have two methods `agree_with_flags` and `defer_flags`.